//bindings/pydrake/visualization:py/model_visualizer_test test_reload failure in CI

### What happened?

 **//bindings/pydrake/visualization:py/model_visualizer_test**  failed in CI:

Aug 7 (x2)
https://drake-jenkins.csail.mit.edu/view/Continuous%20Production/job/mac-arm-monterey-clang-bazel-continuous-release/1101/
https://drake-jenkins.csail.mit.edu/view/Continuous%20Production/job/mac-arm-monterey-clang-bazel-continuous-release/1102/

Aug 15 
https://drake-jenkins.csail.mit.edu/view/Continuous%20Production/job/mac-arm-ventura-clang-bazel-continuous-release/345/

### Version

_No response_

### What operating system are you using?

Other

### What installation option are you using?

_No response_

### Relevant log output

```shell
======================================================================
FAIL [1.205s]: test_reload (model_visualizer_test.TestModelVisualizer.test_reload)
Checks that the _reload() function does not crash.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/admin/workspace/mac-arm-monterey-clang-bazel-continuous-release/_bazel_admin/d8209b58b645930155068a99d03380e4/sandbox/darwin-sandbox/7614/execroot/drake/bazel-out/darwin_arm64-opt/bin/bindings/pydrake/visualization/py/model_visualizer_test.runfiles/drake/bindings/pydrake/visualization/test/model_visualizer_test.py", line 229, in test_reload
    self.assertEqual(meshcat.GetButtonClicks(button), 1)
AssertionError: 0 != 1

----------------------------------------------------------------------
```

--- a/bindings/pydrake/visualization/test/model_visualizer_test.py
+++ b/bindings/pydrake/visualization/test/model_visualizer_test.py
@@ -226,6 +226,12 @@ class TestModelVisualizer(unittest.TestCase):
             cli,
             f"--ws_url={meshcat.ws_url()}",
             f"--send_message={message}"])
+
+        # Wait up to 5 seconds for the button click to be processed.
+        for _ in range(500):
+            if meshcat.GetButtonClicks(button) > 0:
+                break
+            time.sleep(1 / 100)
         self.assertEqual(meshcat.GetButtonClicks(button), 1)
 
         # Run once. If a reload() happened, the diagram will have changed out.

